openshift/ci-operator/generate-ci-config.sh

```diff
@@ -41,6 +41,20 @@ EOF
   done
 }
 
+function generate_cron_expression {
+  if [[ "$branch" == "knative-v0.25.3" ]]; then
+    echo '0 1 * * 1-5'
+  elif [[ "$branch" == "knative-v0.26" ]]; then
+    echo '0 3 * * 1-5'
+  elif [[ "$branch" == "knative-v1.0" ]]; then
+    echo '0 5 * * 1-5'
+  elif [[ "$branch" == "knative-v1.1" ]]; then
+    echo '0 7 * * 1-5'
+  elif [[ "$branch" == "knative-v1.2" ]]; then
+    echo '0 9 * * 1-5'
+  fi
+}
+
```

Comment:
One idea to avoid hard-coding is to use the last number of minor version for the hour:

`knative-v0.25.3` -> `0 5 * * 1-5`
`knative-v0.26` -> `0 6 * * 1-5`
`knative-v1.0` -> `0 0 * * 1-5`
`knative-v1.1` -> `0 1 * * 1-5`
`knative-v1.2` -> `0 2 ** 1-5`

Reply:
If we don't do this, can we update docs?
https://github.com/openshift/knative-eventing#release-schedule

Reply:
Update with what? Sorry I didn't understand

Reply:
If we cut another branch, `knative-v1.3` this doesn't produce any cron expression, so when we cut a new branch or remove an old one we need to update this logic here

Reply:
Or we're missing an else branch.

Reply:
By default it will fallback to the old cron expression. We would need to touch the file though, you are right.

```
knative-v0.25.3 -> 0 5 * * 1-5
knative-v0.26 -> 0 6 * * 1-5
knative-v1.0 -> 0 0 * * 1-5
knative-v1.1 -> 0 1 * * 1-5
knative-v1.2 -> 0 2 ** 1-5
```

This is good, but just adds Bash complexity IMO, which I am not a very big fan of.

Reply:
Added some docs in README now

Reply:
cc @pierDipi